### ISSUE TITLE
I started getting alerts from dropbox again within the last few days.

### ISSUE THREAD

**Issue by kobejohn** [dimaryaz/dropbox_ext4#2]
Thank you for this repo. It has worked for me until this week (using BTRFS)

I guess it's time to leave Dropbox (Pro) after almost 10 years.

**Comment by timonsku** [comment]
I didn't try it out until today but I also still get the message after installing the fix.
I'm running ext4 with ecryptfs.

**Comment by sanmai** [comment]
Guess this commit may shine some light:
https://github.com/dark/dropbox-filesystem-fix/commit/137d82a4f0c293c58396981b031f51a0c2368b33

**Comment by kobejohn** [comment]
Yes. it looks like dropbox is investing resources into bypassing these workarounds instead of just letting us use encrypted ext4, btrfs, etc. It seems there is something else to their decision - it can't just be the file system, can it?

**Comment by sanmai** [comment]
Why won't they bother to explain it if it so? Show us some benchmarks where ext4 works a lot better. Probably all we can understand these arguments. Still this ordeal makes me think the reasons are of non-technical nature. Would glad to be mistaken.

**Comment by kobejohn** [comment]
> Still this ordeal makes me think the reasons are of non-technical nature. Would glad to be mistaken.

Same. I am in the process of setting up nextcloud for myself and relatives.

**Comment by timonsku** [comment]
I'm pretty certain its a very mundane decision of 'lets make our lifes slightly simpler and only allow these major file systems' after seeing that only 0.5% of their users are using something else.

**Comment by kobejohn** [comment]
That sounds good but it doesn’t match up with my (limited) observation. The
only problems I find searching for Dropbox and btrfs are this issue. And
why no encryption?

**Comment by sanmai** [comment]
Reading `man 2 fallocate` I couldn't not notice that there a some features that are ext4-specific, like `FALLOC_FL_INSERT_RANGE` or `FALLOC_FL_PUNCH_HOLE`. If that's the reason, it is somewhat legit. What more, [this is about what Dropbox is checking for](https://github.com/dark/dropbox-filesystem-fix/commit/137d82a4f0c293c58396981b031f51a0c2368b33) nowadays. 

But they do not check for these specific features as of now, so if as of yet you're getting warnings from Dropbox, you're really in trouble. At some point in the future Dropbox will actually stop working, no matter what filesystem detection workarounds you use.

Good news is XFS is on par with ext4 on these features. Now I wonder why won't they declare support for both these filesystem. This would sound like a very credible explanation.

**Comment by timonsku** [comment]
The same should go for cryptfs though when used with ext4 but they specifically disallow that aswell, only full disk encryption is ok.
